nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/PutHDFS.java

```diff
@@ -466,8 +466,13 @@ public void process(InputStream in) throws IOException {
                     Optional<GSSException> causeOptional = findCause(e, GSSException.class, gsse -> GSSException.NO_CRED == gsse.getMajor());
                     if (causeOptional.isPresent()) {
                         getLogger().warn("An error occurred while connecting to HDFS. "
-                                        + "Rolling back session, and penalizing flow file {}",
-                                new Object[] {putFlowFile.getAttribute(CoreAttributes.UUID.key()), causeOptional.get()});
+                                        + "Rolling back session, resetting HDFS resources, and penalizing flow file {}",
```

Comment:
The catch branches can be extended with the GSSRuntimeException with the following way

```
} catch (final GSSRuntimeException e) {
                    getLogger().error("An error occurred while connecting to HDFS. "
                                    + "Rolling back session, and penalizing flow file {}",
                            putFlowFile.getAttribute(CoreAttributes.UUID.key()), e.getCause());
                    getLogger().warn("Resetting HDFS resouces.");
                    try {
                        hdfsResources.set(resetHDFSResources(getConfigLocations(context), context));
                    } catch (IOException ex) {
                        getLogger().debug("Tried to reset HDFS resources when caught GSSException, but IOException was thrown...", ex);
                        throw new RuntimeException(ex);
                    }
                    session.rollback(true);
                } catch (final IOException e) {
                    getLogger().error("Failed to access HDFS due to {}", new Object[]{e});
                    session.transfer(putFlowFile, getFailureRelationship());
                }
```